lib/RexStanSettings.php

```diff
@@ -0,0 +1,250 @@
+ *
+ *   - class->getValue()
+ *     Daten einlesen für ausgewählte Felder aus phpstan-Konfigurationsdateien
+ *       a) user_config.neon
+ *       b) default_config.neon (= Fallback)
+ *     Für alle anderen via rex_config
+ *   - class->save()
+ *     Daten werden für ausgewählte Felder in user_config.neon gespeichert.
+ *     Alle anderen via rex_config
+ *   - class->init()
+ *     Baut die Eingabefelder in das Formular ein
+ *     Bereitet Daten für die Felder auf
+ *
+ * Immer mal wieder anzupassende Auswahllisten (z.B. phpVersionen, Extensons)
+ * sind im vorderen Teil der Klasse zu finden. Init() bereitet daraus
+ * Selects auf.
+ *
+ * Formularausgabe:
+ *      $form = RexStanSettings::factory( 'rexstan' );
+ *      $fragment = new \rex_fragment();
+ *      $fragment->setVar('class', 'edit', false);
+ *      $fragment->setVar('title', 'titel', false);
+ *      $fragment->setVar('body', $form->get(), false);
+ *      echo $fragment->parse('core/page/section.php');.
+ */
+
+namespace redaxo\phpstan;
+
+use rex_addon;
+use rex_config;
+use rex_config_form;
+use rex_extension;
+use rex_extension_point;
+use rex_path;
+use RexStanUserConfig;
+
+use function is_string;
+
+final class RexStanSettings extends rex_config_form
+{
+    /**
+     * Listen der derzeit verfügbaren phpstan-Extensions ... ['label' => 'pfad_im_addon'].
+     *
+     * @var array<string>
+     */
+    private static $phpstanExtensions = [
+        'Bleeding-Edge' => 'vendor/phpstan/phpstan/conf/bleedingEdge.neon',
+        'Strict-Mode' => 'vendor/phpstan/phpstan-strict-rules/rules.neon',
+        'Deprecation Warnings' => 'vendor/phpstan/phpstan-deprecation-rules/rules.neon',
+        'PHPUnit' => 'vendor/phpstan/phpstan-phpunit/rules.neon',
+        'phpstan-dba' => 'lib/phpstan-dba.neon',
+        'cognitive complexity' => 'lib/cognitive-complexity.neon',
+    ];
+
+    /**
+     * ... und der Links zu Erläuterungen. ['label' => 'url'].
+     *
+     * @var array<string>
+     */
+    private static $phpstanExtensionDocLinks = [
+        'Bleeding-Edge' => 'https://phpstan.org/blog/what-is-bleeding-edge',
+        'Strict-Mode' => 'https://github.com/phpstan/phpstan-strict-rules#readme',
+        'Deprecation Warnings' => 'https://github.com/phpstan/phpstan-deprecation-rules#readme">Deprecation-Warnings',
+        'PHPUnit' => 'https://github.com/phpstan/phpstan-phpunit#readme">PHPUnit',
+        'phpstan-dba' => 'https://staabm.github.io/archive.html#phpstan-dba',
+        'cognitive complexity' => 'https://tomasvotruba.com/blog/2018/05/21/is-your-code-readable-by-humans-cognitive-complexity-tells-you/',
+    ];
+
+    /**
+     * Liste der derzeit relevanten PHP-Versionen. [versionId => 'label'].
+     *
+     * @var array<string>
+     */
+    private static $phpVersionList = [
+        70300 => '7.3 [Mindestanforderung für REDAXO]',
+        70400 => '7.4',
+        80000 => '8.0',
+        80100 => '8.1',
+        80200 => '8.2',
```

Comment:
die php-version-id die man in die phpstan-config reinschreibt sollte keine der ersten bugfix versionen sein ( `.0` oder `.1` oder ähnlich, da diese oftmals in verschiedenen tools nicht supported sind, wg. kinderkrankheiten. Beispiel: symfony erfordert mindestens 8.0.2 oder 7.2.5 oder ähnlich).

ist nicht wichtig dass es die allerneusten sind

 ```suggestion
        70333 => '7.3 [Mindestanforderung für REDAXO]',
        70430 => '7.4',
        80022 => '8.0',
        80109 => '8.1',
        80200 => '8.2',
```